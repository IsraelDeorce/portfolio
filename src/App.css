.App {}

.page-body {
  max-width: 700px;
  margin: auto;
  padding-top: 64px;
}
