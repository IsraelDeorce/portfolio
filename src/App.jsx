import { useSelector } from 'react-redux'
import { HashRouter as Router, Routes, Route } from 'react-router-dom'
import { ThemeProvider } from 'styled-components'
import './i18n'
import { Portfolio, Error, Blog } from './pages'
import { Navbar } from './components'
import Themes from './utils/themes'
import * as S from './styles'

function App() {
  const { theme } = useSelector((state) => state.session)

  return (
    <Router basename='/' hashType='noslash'>
      <ThemeProvider theme={Themes[theme]}>
        <S.App id='app'>
          <Navbar />
          <Routes>
            <Route path='/' element={<Portfolio />} />
            <Route path='blog' element={<Blog />} />
            <Route path='blog/big-o' element={<div />} />
            <Route path='blog/data-structures' element={<div />} />
            <Route path='*' element={<Error />} />
          </Routes>
        </S.App>
      </ThemeProvider>
    </Router >
  )
}

export default App
