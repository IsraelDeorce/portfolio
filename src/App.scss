.App {}

.isDisabled {
  cursor: not-allowed;
  opacity: 0.5;
}