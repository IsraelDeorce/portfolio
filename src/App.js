import { BrowserRouter as Router, Switch, Route } from 'react-router-dom'
import { Provider } from 'react-redux'
import store from './store'
import { NavBar } from './components'
import { Home, Error } from './pages'

import './App.scss'

function App() {
  return (
    <Provider store={store}>
      <Router>
        <div id='app'>
          <NavBar />
          <div>
            <Switch>
              <Route exact path='/' component={Home} />
              <Route path='*' component={Error} />
            </Switch>
          </div>
        </div>
      </Router >
    </Provider>
  )
}

export default App
