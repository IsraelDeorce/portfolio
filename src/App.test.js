import { shallow } from 'enzyme'
import App from './App'

describe('<App/>', () => {
  it('renders learn react link', () => {
    console.log(shallow(<App/>).debug())
    expect(true).toBe(true)
  })
})
