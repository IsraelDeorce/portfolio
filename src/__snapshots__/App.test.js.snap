// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> Snapshot renders without crashing using shallow 1`] = `
<HashRouter
  basename="/"
  hashType="noslash"
>
  <Component
    theme={
      Object {
        "c_background": "#1E1E1E",
        "c_brackets": "#D4D4C8",
        "c_component": "#4EC9B0",
        "c_function": "#DCDCAA",
        "c_importExport": "#C586C0",
        "c_menu": "#252526",
        "c_number": "#A7CEA8",
        "c_string": "#CE9078",
        "c_tab": "#3C3C3C",
        "c_varDeclaration": "#4FC1DB",
        "c_varProperty": "#9CDCFE",
        "c_varType": "#569CD6",
      }
    }
  >
    <styled.div
      id="app"
    >
      <Navbar />
      <Routes>
        <Route
          element={<Portfolio />}
          path="/"
        />
        <Route
          element={<Blog />}
          path="blog"
        />
        <Route
          element={<div />}
          path="blog/big-o"
        />
        <Route
          element={<div />}
          path="blog/data-structures"
        />
        <Route
          element={<Error />}
          path="*"
        />
      </Routes>
    </styled.div>
  </Component>
</HashRouter>
`;
